presto-main/src/main/java/io/prestosql/sql/rewrite/ShowStatsRewrite.java

```diff
@@ -154,9 +156,19 @@ protected Node visitShowStats(ShowStats node, Void context)
                 throw rewriteAccessDeniedException(e);
             }
 
+            for (ColumnHandle column : columnHandles.values()) {
+                ColumnMetadata columnMetadata = metadata.getColumnMetadata(session, tableHandle, column);
+                if (!accessControl.getColumnMasks(session.toSecurityContext(), tableName, columnMetadata.getName(), columnMetadata.getType()).isEmpty()) {
+                    throw new PrestoException(NOT_SUPPORTED, "SHOW STATS for table with column masking is not supported: " + columnMetadata.getName());
```

Comment:
This should be verified only for the columns requested.

Ie if i `SHOW STATS FOR SELECT a FROM t`, then mask on `b` doesn't matter